plugins/com.google.cloud.tools.eclipse.appengine.facets.test/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineFlexFacetTest.java

```diff
@@ -38,26 +37,39 @@ public void testFlexFacetDoesNotExists() {
         ProjectFacetsManager.isProjectFacetDefined("com.google.cloud.tools.eclipse.appengine.facets.flex"));
   }
 
-  public void testFlexFacetExists() {
-    Assert.assertTrue(
-        ProjectFacetsManager.isProjectFacetDefined("com.google.cloud.tools.eclipse.appengine.facets.flex"));
-  }
-
+  @Test
   public void testHasAppEngineFacet_withFacet() {
+    if (!ProjectFacetsManager.isProjectFacetDefined(
```

Comment:
This is what the Assume class in Junit is designed for:
http://junit.sourceforge.net/javadoc/org/junit/Assume.html
It lets the test runner notate the tests as skipped rather than passed.